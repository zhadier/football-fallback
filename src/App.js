import { Routes, Route } from 'react-router-dom';
import TeamDetails from './components/TeamDetails';
import Homepage from './components/Homepage';

const App = () => (
  <Routes>
    <Route path="/" element={<Homepage />} />
    <Route path="/teamdetails/:id" element={<TeamDetails />} />
  </Routes>
);

export default App;
