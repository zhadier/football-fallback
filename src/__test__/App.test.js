import React from 'react';
import { setupServer } from 'msw/node';
import {
  waitFor, render, fireEvent, screen, cleanup,
} from './test-utils';
import App from '../App';
import '@testing-library/jest-dom';
import handlers from './mock/handles';

const server = setupServer(...handlers);

afterEach(() => cleanup(), server.resetHandlers());

beforeAll(() => server.listen({
  onUnhandledRequest: 'error',
}));

afterAll(() => server.close());

it('App Renders Correctly', async () => {
  const { container } = render(<App />);
  await waitFor(() => screen.getByText('Premier League'));
  expect(container).toMatchSnapshot();
});

it('Clicking on a team takes you to its details page', async () => {
  render(<App />);
  await waitFor(() => screen.getByText('Liverpool'));
  fireEvent.click(screen.getByText('Liverpool'));
  expect(screen.getByText('All Stats')).toBeInTheDocument();
});

it('Clicking on < takes you back to the home page', async () => {
  render(<App />);
  fireEvent.click(screen.getByTestId('back'));
  expect(screen.getByText('Team rankings')).toBeInTheDocument();
});
