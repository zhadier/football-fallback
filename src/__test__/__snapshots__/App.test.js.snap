// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<div>
  <header
    class="header"
  >
    <input
      class="search"
      placeholder="🔍 Search"
      type="text"
      value=""
    />
    <div
      class="icons-container"
    >
      <svg
        class="header-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 12V6c0-2.217-1.785-4.021-3.979-4.021a.933.933 0 0 0-.209.025A4.006 4.006 0 0 0 8 6v6c0 2.206 1.794 4 4 4s4-1.794 4-4zm-6 0V6c0-1.103.897-2 2-2a.89.89 0 0 0 .163-.015C13.188 4.06 14 4.935 14 6v6c0 1.103-.897 2-2 2s-2-.897-2-2z"
        />
        <path
          d="M6 12H4c0 4.072 3.061 7.436 7 7.931V22h2v-2.069c3.939-.495 7-3.858 7-7.931h-2c0 3.309-2.691 6-6 6s-6-2.691-6-6z"
        />
      </svg>
      <svg
        class="header-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </div>
  </header>
  <div
    class="league-info"
  >
    <img
      alt="league"
      src="https://media.api-sports.io/football/leagues/2.png"
    />
    <div
      class="league-description"
    >
      <h2
        class="league-name"
      >
        Premier League
      </h2>
      <p>
        England
      </p>
      <p>
        2021 - 2022
      </p>
    </div>
  </div>
  <ul
    class="teams-section"
  >
    <h2
      class="teams-header"
    >
      Team rankings
    </h2>
    <li
      class="team-container"
      id="team-40"
    >
      <a
        data-testid="test-40"
        href="/teamdetails/40"
      >
        <svg
          class="right-arrow-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fill-rule="evenodd"
          />
        </svg>
        <img
          alt="team-logo"
          src="https://media.api-sports.io/football/teams/40.png"
        />
        <div
          class="team-desc-container"
        >
          <h2>
            Liverpool
          </h2>
          <p>
            Rank :
             
            1
          </p>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
